File: TEST/blog.go
type SECTION struct {
    Id uint64 `db:"Id"`;
    Number uint64 `db:"Number"`;
    Name string `db:"Name"`;
    Text string `db:"Text"`;
    Image string `db:"Image"`;
    ArticleIdList string `db:"ArticleIdList"`;
    ImageIndex uint64;
}

// ~~

type USER struct {
    Id uint64 `db:"Id"`;
    FirstName string `db:"FirstName"`;
    LastName string `db:"LastName"`;
    Email string `db:"Email"`;
    Pseudonym string `db:"Pseudonym"`;
    Password string `db:"Password"`;
    Phone string `db:"Phone"`;
    Street string `db:"Street"`;
    City string `db:"City"`;
    Code string `db:"Code"`;
    Region string `db:"Region"`;
    Country string `db:"Country"`;
    Company string `db:"Company"`;
    ItIsAdministrator bool `db:"ItIsAdministrator"`;
}

// ~~

type ARTICLE struct {
    Id uint64 `db:"Id"`;
    SectionId uint64 `db:"SectionId"`;
    UserId uint64 `db:"UserId"`;
    Title string `db:"Title"`;
    Text string `db:"Text"`;
    Image string `db:"Image"`;
    Date string `db:"Date"`;
    Section * SECTION;
    User * USER;
    ImageIndex uint64;
}

// ~~

type COMMENT struct {
    Id uint64 `db:"Id"`;
    ArticleId uint64 `db:"ArticleId"`;
    UserId uint64 `db:"UserId"`;
    Text string `db:"Text"`;
    DateTime string `db:"DateTime"`;
    Article * ARTICLE;
    User * USER;
}

// ~~

type SUBSCRIBER struct {
    Id uint64 `db:"Id"`;
    Name string `db:"Name"`;
    Email string `db:"Email"`;
}

// ~~

File: SAMPLE/blog.rs
struct SECTION
{
    Id : u64,
    Number : u64,
    Name : String,
    Text : String,
    Image : String,
    ImageIndex : u64
}

struct USER
{
    Id : u64,
    FirstName : String,
    LastName : String,
    Email : String,
    Pseudonym : String,
    Password : String,
    Phone : String,
    Street : String,
    City : String,
    Code : String,
    Region : String,
    Country : String,
    Company : String,
    ItIsAdministrator : bool
}

struct ARTICLE
{
    Id : u64,
    SectionId : u64,
    UserId : u64,
    Title : String,
    Text : String,
    Image : String,
    Date : String,
    Section : &SECTION,
    User : &USER,
    ImageIndex : u64
}

struct COMMENT
{
    Id : u64,
    ArticleId : u64,
    UserId : u64,
    Text : String,
    DateTime : String,
    Article : &ARTICLE,
    User : &USER
}

struct SUBSCRIBER
{
    Id : u64,
    Name : String,
    Email : String
}

